module/zfs/ddt.c

```diff
@@ -719,6 +733,15 @@ ddt_lookup(ddt_t *ddt, const blkptr_t *bp, boolean_t add)
 
 	ddt_enter(ddt);
 
+	dde_temp = avl_find(&ddt->ddt_tree, &dde_search, &where);
```

Comment:
Can you add comments here about what's going on, something like: we're removing the newly added entry from our in core DDE tree because we're not interested in growing the DDT at this point; if needed this write will be transformed into a regular write (see zio_ddt_write())